src/write_gcm.c

```diff
@@ -3105,6 +3146,53 @@ static void wg_json_Metric(json_ctx_t *jc,
   wg_json_map_close(jc);
 }
 
+static int wg_metric_value_type_from_wg_payload_t(
+    const wg_payload_t *element) {
+  int metric_value_type = -1;
+  for (int i = 0; i < element->key.num_metadata_entries; ++i) {
+    wg_metadata_entry_t *entry = &element->key.metadata_entries[i];
+    if (strcmp(entry->key, custom_metric_value_type) == 0) {
+      if (strcmp(entry->value.value_text, "INT64") == 0) {
+        metric_value_type = METRIC_VALUE_TYPE_INT64;
+      } else if (strcmp(entry->value.value_text, "DOUBLE") == 0) {
+        metric_value_type = METRIC_VALUE_TYPE_DOUBLE;
+      } else if (strcmp(entry->value.value_text, "BOOL") == 0) {
+        metric_value_type = METRIC_VALUE_TYPE_BOOL;
+      } else {
+        WARNING("write_gcm: unsupported metric value type provided: %s", 
+                entry->value.value_text);
+      }
+      break;
+    }
+  }
+
+  // Failover to sane defaults for each DS type.
+  if(metric_value_type == -1) {
+    switch (element->values[0].ds_type) {
```

Comment:
This switch would need to be a separate helper function that you call from `wg_json_CollectdValues` (e.g., `wg_default_metric_value_type_from_wg_payload_t`).

Reply:
Done, called the helper function `wg_default_metric_value_type_from_ds_type` as I had access to the `ds_type` in all invocations.